cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1172,21 +1199,41 @@ protected Affinity getMergedAffinity() {
             // If there's a rack config, we need to add a podAntiAffinity to spread the brokers among the racks
             builder = builder
                     .editOrNewPodAntiAffinity()
-                        .addNewPreferredDuringSchedulingIgnoredDuringExecution()
-                            .withWeight(100)
-                            .withNewPodAffinityTerm()
-                                .withTopologyKey(rack.getTopologyKey())
-                                .withNewLabelSelector()
-                                    .addToMatchLabels(Labels.STRIMZI_CLUSTER_LABEL, cluster)
-                                    .addToMatchLabels(Labels.STRIMZI_NAME_LABEL, name)
-                                .endLabelSelector()
-                            .endPodAffinityTerm()
-                        .endPreferredDuringSchedulingIgnoredDuringExecution()
+                    .addNewPreferredDuringSchedulingIgnoredDuringExecution()
+                    .withWeight(100)
+                    .withNewPodAffinityTerm()
+                    .withTopologyKey(rack.getTopologyKey())
+                    .withNewLabelSelector()
+                    .addToMatchLabels(Labels.STRIMZI_CLUSTER_LABEL, cluster)
+                    .addToMatchLabels(Labels.STRIMZI_NAME_LABEL, name)
+                    .endLabelSelector()
+                    .endPodAffinityTerm()
+                    .endPreferredDuringSchedulingIgnoredDuringExecution()
                     .endPodAntiAffinity();
         }
         return builder.build();
     }
 
+    protected List<EnvVar> getInitContainerEnvVars() {
+        List<EnvVar> varList = new ArrayList<>();
+        varList.add(buildEnvVarFromFieldRef(ENV_VAR_KAFKA_INIT_NODE_NAME, "spec.nodeName"));
+
+        if (rack != null) {
+            varList.add(buildEnvVar(ENV_VAR_KAFKA_INIT_RACK_TOPOLOGY_KEY, rack.getTopologyKey()));
+        }
+
+        if (isExposedWithNodePort()) {
+            varList.add(buildEnvVar(ENV_VAR_KAFKA_INIT_EXTERNAL_ADDRESS, "TRUE"));
+            varList.add(buildEnvVar(ENV_VAR_KAFKA_INIT_EXTERNAL_ADVERTISED_ADDRESSES, String.join(" ", externalAddresses)));
+        }
+
+        if (templateInitContainerEnvVars != null) {
+            addContainerEnvsToExistingEnvs(varList, templateInitContainerEnvVars);
```

Comment:
If you put a null check in `addContainerEnvsToExistingEnvs()` (on its 2nd parameter) then you can avoid having a null check at every call site.

Reply:
That is a very good point! I will add that in an remove the checks